lib/client/db/dbcmd/dbcmd.go

```diff
@@ -143,20 +151,32 @@ func (c *cliCommandBuilder) getConnectCommand() (*exec.Cmd, error) {
 	return nil, trace.BadParameter("unsupported database protocol: %v", c.db)
 }
 
+func (c *cliCommandBuilder) GetRelativeConnectCommand() (*exec.Cmd, error) {
+	cmd, err := c.GetConnectCommand()
+
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if filepath.IsAbs(cmd.Path) {
+		cmd.Path = filepath.Base(cmd.Path)
+	}
+
+	return cmd, nil
+}
+
 func (c *cliCommandBuilder) getPostgresCommand() *exec.Cmd {
-	return exec.Command(postgresBin,
+	return c.exe.Command(postgresBin,
 		postgres.GetConnString(db.New(c.tc, *c.db, *c.profile, c.rootCluster, c.host, c.port), c.options.noTLS))
 }
 
 func (c *cliCommandBuilder) getCockroachCommand() *exec.Cmd {
 	// If cockroach CLI client is not available, fallback to psql.
 	if _, err := c.exe.LookPath(cockroachBin); err != nil {
-		log.Debugf("Couldn't find %q client in PATH, falling back to %q: %v.",
```

Comment:
Any reason you removed this log entry?

Reply:
Ah, yes, sorry, I forgot to mention that.

That was a dirty workaround on my part. `getCockroachCommand` was the only function in `dbcmd` that called private `log` from the `tsh` package when `dbcmd` was still in that package. Instead of figuring out what's the best way to supply the logger here, I just removed the only log statement.

Is there any specific way of dealing with this problem that you'd recommend? `tsh` and `lib/teleterm/clusters`, the two packages that use `dbcmd` both have access to a logger instance which could be passed to dbcmd. But I noticed that code under `lib/client` tends to create its own instance of logger instead – would that be the way to go?

Reply:
Yes, the way I would do it is:

- Make `log logrus.FieldLogger` a part of `connectionCommandOpt` struct.
- Implement `WithLogger` functional argument and have the caller optionally pass it via opts.
- If it's not passed, initialize the logger to `logrus.NewStandardLogger()` when creating new `cliCommandBuilder`.